### PR TITLE
First fixes for the package review

### PR DESCRIPTION
**Pull request by jorainer**
- This should fix all issues except the subsetting and the class union of `MsExperimentFiles` that includes `NULL`.

### CODE REVIEW

File: R/MsExperiment.R [hunk 1]
```diff
@@ -46,6 +47,13 @@
 #'   `PSM()` objects) can be added as elements to the list stored in
 #'   the `otherData` slot.
 #'
+#' The *length* of a `MsExperiment` is defined by the number of samples (i.e.
+#' the number of rows of the object's `sampleData`). A `MsExperiment` with two
```
Comment (lgatto) [inline R/MsExperiment.R:51]: A*n* `MsExperiment`

File: R/MsExperiment.R [hunk 2]
```diff
@@ -128,8 +136,8 @@
 #'
 #' @section Subsetting and filtering:
 #'
-#' - `[`: `MsExperiment` objects can be subsetted **by samples** with `[, j]`
-#'   where `j` is the index or a logical defining to which samples the data
+#' - `[`: `MsExperiment` objects can be subsetted **by samples** with `[i]`
```
Comment (lgatto) [inline R/MsExperiment.R]: ... can be subset

**Review by lgatto — CHANGES_REQUESTED** [review]

File: R/MsExperiment.R (hunk 2 above)
Comment (jorainer) [reply R/MsExperiment.R]: fixed now

**Comment by jorainer** [comment]
@lgatto all changes have now been done. Additional unit tests were added. Also, I tried to describe the subsetting *issue* better in the documentation and I added also an example describing that.

**Pull request merged**